packages/ember-data/lib/adapters/json-api-adapter.js

```diff
@@ -39,6 +39,55 @@ export default RESTAdapter.extend({
     return hash;
   },
 
+    /**
+    By default the JSONAPIAdapter will send each find request coming from a `store.find`
+    or from accessing a relationship separately to the server. If your server supports passing
+    ids as a query string, you can set coalesceFindRequests to true to coalesce all find requests
+    within a single runloop.
+
+    For example, if you have an initial payload of:
+
+    ```javascript
+    {
+      post: {
+        id: 1,
+        comments: [1, 2]
+      }
+    }
+    ```
+
+    By default calling `post.get('comments')` will trigger the following requests(assuming the
+    comments haven't been loaded before):
+
+    ```
+    GET /comments/1
+    GET /comments/2
+    ```
+
+    If you set coalesceFindRequests to `true` it will instead trigger the following request:
+
+    ```
+    GET /comments?filter[id]=1,2
+    ```
+
+    Setting coalesceFindRequests to `true` also works for `store.find` requests and `belongsTo`
+    relationships accessed within the same runloop. If you set `coalesceFindRequests: true`
+
+    ```javascript
+    store.findRecord('comment', 1);
+    store.findRecord('comment', 2);
+    ```
+
+    will also send a request to: `GET /comments?filter[id]=1,2`
+
+    Note: Requests coalescing rely on URL building strategy. So if you override `buildURL` in your app
+    `groupRecordsForFindMany` more likely should be overridden as well in order for coalescing to work.
+
+    @property coalesceFindRequests
+    @type {boolean}
+  */
+  coalesceFindRequests: false,
+  
```

Comment:
Thanks for the steps, that's a lot easier than what I was trying with rebase and other things I found online!